src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java

```diff
@@ -32,223 +31,160 @@
+		String [] handle_impl_files = {"__init__.py", "AbstractHandleImpl.py",
+				"AbstractHandleServer.py", "authclient.py", "baseclient.py"};
+		for (String file_name : handle_impl_files) {
+			FileUtils.copyURLToFile(new URL(handle_repo_prefix + file_name),
+					handle_dir.resolve(file_name).toFile());
 		}
-		
-		final Ini ini = new Ini();
-		final Section hm = ini.add("HandleMngr");
-		hm.add("handle-service-url", "http://localhost:" + handleServicePort);
-		hm.add("service-host", "localhost");
-		hm.add("service-port", "" + handleManagerPort);
-		hm.add("auth-service-url", authServiceURL.toString());
-		hm.add("admin-token", shockAdminToken.getToken());
-		hm.add("allowed-users", allowedUser);
-		
-		ini.store(iniFile);
-		return iniFile;
+
+		Path handle_utils_dir = handle_dir.resolve("Utils");
+		Files.createDirectories(handle_utils_dir);
+		String [] handle_util_files = {"__init__.py", "Handler.py", "MongoUtil.py",
+				"ShockUtil.py", "TokenCache.py"};
+		for (String file_name : handle_util_files) {
+			FileUtils.copyURLToFile(new URL(handle_repo_prefix + "Utils/" + file_name),
+					handle_utils_dir.resolve(file_name).toFile());
+		}
+
+		Path biokbase_dir = lib_root.resolve("biokbase");
+		Files.createDirectories(biokbase_dir);
+
+		String biokbase_repo_prefix = "https://raw.githubusercontent.com/kbase/sdkbase2/python/";
+		String [] biokbase_files = {"log.py"};
+		for (String file_name : biokbase_files) {
+			FileUtils.copyURLToFile(new URL(biokbase_repo_prefix + file_name),
+					biokbase_dir.resolve(file_name).toFile());
+		}
+
 	}
-	
+
 	private File createHandleServiceDeployCfg(
-			final MySQLController mysql,
+			final MongoController mongo,
 			final String shockHost,
-			final URL authServiceURL) throws IOException {
+			final URL authURL,
+			final AuthToken shockAdminToken,
+			final String handleAdminRole) throws IOException {
 		final File iniFile = tempDir.resolve("handleService.cfg").toFile();
 		if (iniFile.exists()) {
 			iniFile.delete();
 		}
-		
+
 		final Ini ini = new Ini();
 		final Section hs = ini.add(HANDLE_SERVICE_NAME);
 		hs.add("self-url", "http://localhost:" + handleServicePort);
 		hs.add("service-port", "" + handleServicePort);
 		hs.add("service-host", "localhost");
+		URL authServiceURL = new URL(authURL.toString() + "/api/legacy/KBase/Sessions/Login");
 		hs.add("auth-service-url", authServiceURL.toString());
-		hs.add("default-shock-server", shockHost);
-		
-		hs.add("mysql-host", "127.0.0.1");
-		hs.add("mysql-port", "" + mysql.getServerPort());
-		hs.add("mysql-user", USER);
-		hs.add("mysql-pass", PWD);
-		hs.add("data-source", "dbi:mysql:" + DB);
-		
+		hs.add("auth-url", authURL.toString());
+		hs.add("shock-url", shockHost);
+		hs.add("admin-token", shockAdminToken.getToken().toString());
+		hs.add("admin-roles", handleAdminRole);
+
+		hs.add("mongo-host", "127.0.0.1");
+		hs.add("mongo-port", "" + mongo.getServerPort());
+		hs.add("mongo-database", "handle_db");
```

Comment:
Can you make this a constructor param so it can be configured or make is something that's less likely to have a name collision? Maybe `handle_controller_test_handle_db` or something

Reply:
👍